docs/book/src/cookbook/cookbook.md

```diff
@@ -0,0 +1,32 @@
+# Cookbook
+
+A curated collection of operator development patterns, use-cases and generic\
+recommendations.
+
+## Rationale
+
+While the tutorial part of this book provides onboarding into the operator
+realm, real world use cases often surpass the tutorial example(s). Hence we
+want to provide a place to share common implementation patterns in order
+to enable other users, may they be operator experts or novices, a place to both
+share and validate common approaches they encounter while developing operators.
+
+## Recipies
+
+<aside class="note">
+
+<h1>Following Along</h1>
+
+Note that code examples of cookbook entries live as literate Go files in the
+book source directory:
+[docs/book/src/cookbook/testdata][cookbook-sources]. You may find both snippets
+and fully runnable projects among the examples.
+
+[cookbook-sources]: https://github.com/kubernetes-sigs/kubebuilder/tree/master/docs/book/src/cookbook/testdata
+
+</aside>
+
+*If you are reading this and you are wondering where the entries are: there are
```

Comment:
I like the idea of tips, tricks and advices one should know when writes an operator! I am not sure though I understand what a valid/expected contribution might be considered in this though `testdata` -- which by the way the word `test` is confusing me in this context. If you could come up with an example of a such a contribution already, that would be great.

Reply:
I borrowed this section from another part of the book as I thought it has been established as the idiomatic way to provide examples throughout the docs.

I agree that an example would be helpful. Right now, especially after diving into https://github.com/kubernetes-sigs/kubebuilder/pull/1477#discussion_r415032376, I am tending to start with a pure-test example which would only use embedded code snippets if it has to at all.

Reply:
FYI on the testdata directory: https://github.com/kubernetes-sigs/kubebuilder/pull/1477#discussion_r416165222